core/app/models/workarea/search/storefront/product.rb

```diff
@@ -38,7 +38,8 @@ def keywords
           {
             catalog_id: clean_for_keywords(catalog_id),
             sku: sku.map { |s| clean_for_keywords(s) },
-            name: clean_for_keywords(model.name)
+            name: clean_for_keywords(model.name),
+            tags: model.tag_list
```

Comment:
Seems like this should just be an array so we can provide exact matching without worrying about commas.